pkg/node/manager/manager.go

```diff
@@ -318,7 +319,12 @@ func (m *Manager) legacyNodeIpBehavior() bool {
 	// ipcache. This resulted in a behavioral change. New deployments will
 	// provide this behavior out of the gate, existing deployments will
 	// have to opt into this by enabling remote-node identities.
-	return !m.conf.EncryptionEnabled() && !m.conf.RemoteNodeIdentitiesEnabled()
+
+	// We also need to expose the internalIP to the ipcache if node encryption
+	// is enabled (to store the SPI for nodes) or if pod encryption is enabled
+	// in tunneling mode (to store the SPI for pod->remote node).
+	return !m.conf.NodeEncryptionEnabled() && !(m.conf.EncryptionEnabled() &&
+		m.conf.TunnelingEnabled()) && !m.conf.RemoteNodeIdentitiesEnabled()
```

Comment:
Still having a bit of trouble parsing this. Maybe breaking it down into new lines and annotating each line might help? Up to you.

```suggestion
	return !m.conf.NodeEncryptionEnabled() /* ... */
	  && !(m.conf.EncryptionEnabled() && m.conf.TunnelingEnabled()) /* ... */
	  && !m.conf.RemoteNodeIdentitiesEnabled()
```

Reply:
I split the condition to avoid super long lines. See updated commit.